app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetUserAccountUseCase.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2017.
+ *
+ * This file is part of QA App.
+ *
+ *  Health Network QIS App is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Health Network QIS App is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.eyeseetea.malariacare.domain.usecase;
+
+import org.eyeseetea.malariacare.data.NetworkStrategy;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.domain.boundary.IUserAccountRepository;
+import org.eyeseetea.malariacare.domain.boundary.executors.IAsyncExecutor;
+import org.eyeseetea.malariacare.domain.boundary.executors.IMainExecutor;
+import org.eyeseetea.malariacare.domain.entity.UserAccount;
+
+public class GetUserAccountUseCase implements UseCase {
+
+    public interface Callback {
+        void onSuccess(UserAccount user);
+
+        void onError(UserAccount user);
+    }
+
+    private final IAsyncExecutor mAsyncExecutor;
+    private final IMainExecutor mMainExecutor;
+    private final IUserAccountRepository userAccountRepository;
+    private Callback mCallback;
+
+    public GetUserAccountUseCase(
+            IAsyncExecutor asyncExecutor,
+            IMainExecutor mainExecutor,
+            IUserAccountRepository userAccountRepository) {
+        this.mAsyncExecutor = asyncExecutor;
+        this.mMainExecutor = mainExecutor;
+        this.userAccountRepository = userAccountRepository;
+    }
+
+    public void execute(final Callback callback) {
+        this.mCallback = callback;
+        mAsyncExecutor.run(this);
+    }
+
+    @Override
+    public void run() {
+        //get logged user
+        UserFilter userFilter = new UserFilter("", true);
```

Comment:
In this use case there are some improvements to realize:
- An UseCase should not invoke to  PreferencesState.getInstance().setUserAccept
- UserAccept is a concept of the user then we should be in UserAccount and in database
- The logic to set UserAccept to false should be inside UserAccount entity because is a Bussiness Rule logic and the values to decide are inside UserAccount and we should create a unit test to validate the logic
- An UseCase should not execute the repository first with LocalFirst and then with NetworkFirst
- UserFilter should be passed from outside to UseCase